### PR TITLE
Security checklist: File system permissions

### PR DESCRIPTION
**Pull request by glye**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | All
| Edition       | All

Added info on file system permissions, and some related improvements, to the "Underlying stack" section.

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [x] ~Code samples are working~
- [x] ~PHP code samples have been fixed with PHP CS fixer~
- [x] ~Added link to this PR in relevant JIRA ticket or code PR~

### CODE REVIEW

**Review by adriendupuis — APPROVED** [review]
Another topic for another PR: About DB, it could be interesting to have an example of `GRANT` allowing only the "operations needed".

File: docs/infrastructure_and_maintenance/security/security_checklist.md [hunk 1]
```diff
@@ -190,7 +190,10 @@ Once you have properly configured secure user roles and permissions, to avoid ex
 
 - Avoid exposing servers on the open internet when not strictly required.
 - Ensure any servers, services, ports and virtual hosts that were opened for testing purposes are shut down before going live.
-- Secure the database with a good password, keys, firewall, etc. Ensure that the database user used by the web app only has access to do the operations needed by [[= product_name =]]. The Data Definition Language (DDL) commands (create, alter, drop, truncate, comment) are not needed for running [[= product_name =]], only for installing and upgrading it. If the web app user does not have these rights, then that reduces the damage that can be done if there is a security breach.
+- Ensure file system permissions are set up such that the web server / PHP user cannot access files it shouldn't be able to read.
```
Comment (adriendupuis) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:193]:
```suggestion
- Ensure file system permissions are set up such that the web server or PHP user cannot access files it shouldn't be able to read.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md [hunk 2]
```diff
@@ -190,7 +190,10 @@ Once you have properly configured secure user roles and permissions, to avoid ex
...
+- Secure the database with a good password, keys, firewall, etc. Optionally, ensure that the database user used by the web app only has access to do the operations needed by [[= product_name =]]. The Data Definition Language (DDL) commands (create, alter, drop, truncate, comment) are not needed for running [[= product_name =]], only for installing and upgrading it. If the web app user does not have these rights, then that reduces the damage that can be done if there is a security breach.
+
+The above steps are not needed when using Ibexa Cloud, where these things are handled by the provider.
```
Comment (adriendupuis) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:196]:
```suggestion
The above steps are not needed when using [[= product_name_cloud =]], where these things are handled by the provider.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md [hunk 3]
```diff
@@ -190,7 +190,10 @@ Once you have properly configured secure user roles and permissions, to avoid ex
 
 - Avoid exposing servers on the open internet when not strictly required.
 - Ensure any servers, services, ports and virtual hosts that were opened for testing purposes are shut down before going live.
-- Secure the database with a good password, keys, firewall, etc. Ensure that the database user used by the web app only has access to do the operations needed by [[= product_name =]]. The Data Definition Language (DDL) commands (create, alter, drop, truncate, comment) are not needed for running [[= product_name =]], only for installing and upgrading it. If the web app user does not have these rights, then that reduces the damage that can be done if there is a security breach.
+- Ensure file system permissions are set up such that the web server or PHP user can't access files it shouldn't be able to read.
```
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:193]:
```suggestion
- Ensure file system permissions are set up in such way that the web server or PHP user can't access files it shouldn't be able to read.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md [hunk 4]
```diff
@@ -190,7 +190,10 @@ Once you have properly configured secure user roles and permissions, to avoid ex
...
+- Secure the database with a good password, keys, firewall, etc. Optionally, ensure that the database user used by the web app only has access to do the operations needed by [[= product_name =]]. The Data Definition Language (DDL) commands (create, alter, drop, truncate, comment) are not needed for running [[= product_name =]], only for installing and upgrading it. If the web app user does not have these rights, then that reduces the damage that can be done if there is a security breach.
```
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:194]:
```suggestion
- Secure the database with a good password, keys, firewall, etc.
Optionally, ensure that the database user used by the web app only has permissions to do the operations needed by [[= product_name =]].
The Data Definition Language (DDL) commands (create, alter, drop, truncate, comment) are only needed for installing and upgrading [[= product_name =]], and not for running it.
Not granting these rights to web app users reduces the damage that can result from a security breach.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/infrastructure_and_maintenance/security/security_checklist.md (hunk 3 above)
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:193]: Ok with me, but then "in such **a** way" please.
```suggestion
- Ensure file system permissions are set up in such a way that the web server or PHP user can't access files it shouldn't be able to read.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md [hunk 5]
```diff
@@ -190,7 +190,13 @@ Once you have properly configured secure user roles and permissions, to avoid ex
 
 - Avoid exposing servers on the open internet when not strictly required.
 - Ensure any servers, services, ports and virtual hosts that were opened for testing purposes are shut down before going live.
-- Secure the database with a good password, keys, firewall, etc. Ensure that the database user used by the web app only has access to do the operations needed by [[= product_name =]]. The Data Definition Language (DDL) commands (create, alter, drop, truncate, comment) are not needed for running [[= product_name =]], only for installing and upgrading it. If the web app user does not have these rights, then that reduces the damage that can be done if there is a security breach.
+- Ensure file system permissions are set up in such a way that the web server or PHP user can't access files it shouldn't be able to read.
```
Comment (glye) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:193]:
```suggestion
- Ensure file system permissions are set up in such a way that the web server or PHP user can't access files they shouldn't be able to read.
```

**Pull request merged**